classes/Kohana/Jam/Field/Integer.php

```diff
@@ -32,5 +32,36 @@ public function set(Jam_Validated $model, $value, $is_changed)
 
 		return $value;
 	}
+	
+	/**
+	 * Like Jam_Field::_default, but it won't convert 0 and 0.0 to empty_value
+	 *
+	 * @param  Jam_Validated $model
+	 * @param  mixed $value
+	 * @return array 1st element the converted value; 2nd element a boolean
+	 * indicating if the value should not be processed further
+	 */
+	protected function _default(Jam_Validated $model, $value)
+	{
+		$return = FALSE;
+
+		$value = $this->run_filters($model, $value);
+
+		// Convert empty values to NULL, if needed
+		if ($this->convert_empty AND empty($value) AND $value !== 0 AND $value !== 0.0 AND $value !== "0" AND $value !== "0.0")
```

Comment:
@sdipchikov Actually we want all empty values including zeros to be converted to the empty value.

```
var_dump(empty("0")); // -> (bool) true
var_dump(empty(0)); // -> (bool) true
var_dump(empty("0.0")); // -> (bool) false
```

1. Why do you want to not convert the zeros? Do you realize you can configure the `empty_value` option on the field instance so you could set it to `0`?
2. The only problem I see with the current code in the field is we don't convert `"0.0"` to the empty value. You could adjust your code to do just that.
3. You should update the tests as well.

:beers: